File: Python/linked-list-stack-test.py
from lib.linked_list_stack import Stack

stack = Stack()
while True:
    print("Operations:\n"
          "1 - Push a number\n"
          "2 - Pop a number\n"
          "3 - Peek last value\n"
          "4 - Current size\n"
          "5 - Exit\n")
    choice = int(input("Enter the choice: "))
    print()
    if choice == 1:
        value = int(input("Enter the value: "))
        print()
        stack.push(value)
    elif choice == 2:
        stack.pop()
    elif choice == 3:
        print(stack.peek())
        print()
    elif choice == 4:
        print(stack.get_size())
        print()
    elif choice == 5:
        break
    else:
        print("Invalid choice!\n")

File: TypeScript/linked-list-stack-test.ts
import { Stack } from "./lib/linked-list-stack"
import * as readLine from "readline-sync";


const stack = new Stack();
while (true) {
    console.log("Operations:\n" +
        "1 - Push a number\n" +
        "2 - Pop a number\n" +
        "3 - Peek last value\n" +
        "4 - Current size\n" +
        "5 - Exit\n"
    );
    const choice = readLine.question("Enter the choice: ");
    console.log();
    if (choice === "1") {
        const value = readLine.question("Enter the value: ")
        console.log();
        stack.push(value);
    } else if (choice === "2") {
        stack.pop();
    } else if (choice === "3") {
        console.log(stack.peek());
        console.log();
    } else if (choice === "4") {
        console.log(stack.getSize());
        console.log();
    } else if (choice === "5") {
        break;
    } else {
        console.log("Invalid choice!\n");
    }
}

